persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/StargateQueryHandler.java

```diff
@@ -260,57 +275,216 @@ private void authorizeByToken(ByteBuffer token, CQLStatement statement) {
           castStatement.keyspace(),
           castStatement.columnFamily());
     } else if (statement instanceof SchemaAlteringStatement) {
-      SchemaAlteringStatement castStatement = (SchemaAlteringStatement) statement;
-      Scope scope = null;
-      String keyspaceName = null;
-      String tableName = null;
-
-      if (statement instanceof CreateTableStatement) {
-        scope = Scope.CREATE;
-        keyspaceName = castStatement.keyspace();
-        tableName = castStatement.columnFamily();
-      } else if (statement instanceof DropTableStatement) {
-        scope = Scope.DROP;
-        keyspaceName = castStatement.keyspace();
-        tableName = castStatement.columnFamily();
-      } else if (statement instanceof AlterTableStatement) {
-        scope = Scope.ALTER;
-        keyspaceName = castStatement.keyspace();
-        tableName = castStatement.columnFamily();
-      } else if (statement instanceof CreateKeyspaceStatement) {
-        scope = Scope.CREATE;
-        keyspaceName = castStatement.keyspace();
-        tableName = null;
-      } else if (statement instanceof DropKeyspaceStatement) {
-        scope = Scope.DROP;
-        keyspaceName = castStatement.keyspace();
-        tableName = null;
-      } else if (statement instanceof AlterKeyspaceStatement) {
-        scope = Scope.ALTER;
-        keyspaceName = castStatement.keyspace();
-        tableName = null;
+      authorizeSchemaAlteringStatement(statement, authToken, authorization);
+    } else if (statement instanceof AuthorizationStatement) {
+      authorizeAuthorizationStatement(statement, authToken, authorization);
+    } else if (statement instanceof AuthenticationStatement) {
+      authorizeAuthenticationStatement(statement, authToken, authorization);
+    }
```

Comment:
This should have a final `else` and throw for unhandled statement types.

Reply:
Probably a good idea, adding

Reply:
`java.lang.UnsupportedOperationException` sound right?

Reply:
yes, that seems fine.